From ff7a83ad38611fd08b7ada1f3807d0a8257b66d4 Mon Sep 17 00:00:00 2001
From: czou <cao.zou@windriver.com>
Date: Fri, 28 Oct 2016 15:47:40 +0800
Subject: [PATCH 1/3] crypto: caam - fix memleak in sm_keystore_slot_expor

It fixed the kmemleak as follow:

unreferenced object 0xa91cd180 (size 64):
  comm "swapper/0", pid 1, jiffies 4294938000 (age 168.180s)
  hex dump (first 32 bytes):
    0f 0e 0d 0c 0b 0a 09 08 00 00 00 00 00 00 00 00 ................
    00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ................
  backtrace:
    [<8084cda8>] kmemleak_alloc+0x40/0x74
    [<80142520>] __kmalloc+0x1fc/0x35c
    [<806c4678>] sm_keystore_slot_export+0x60/0x328
    [<806c5248>] caam_sm_example_init+0x5bc/0xa6c
    [<80c17a24>] caam_sm_test_init+0x50/0x60
    [<80009774>] do_one_initcall+0x114/0x1c4
    [<80bc3eb4>] kernel_init_freeable+0x190/0x28c
    [<8084bd60>] kernel_init+0x1c/0xf4
    [<8000f3c8>] ret_from_fork+0x14/0x2c
    [<ffffffff>] 0xffffffff

Signed-off-by: czou <cao.zou@windriver.com>
[Quanyang: Add kfree to function sm_keystore_slot_export]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/crypto/caam/sm_store.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/crypto/caam/sm_store.c b/drivers/crypto/caam/sm_store.c
index 1fa92c4..c54a225 100644
--- a/drivers/crypto/caam/sm_store.c
+++ b/drivers/crypto/caam/sm_store.c
@@ -912,6 +912,7 @@ int sm_keystore_slot_export(struct device *dev, u32 unit, u32 slot, u8 keycolor,
 			 DMA_FROM_DEVICE);
 	dma_unmap_single(dev, keymod_dma, SECMEM_KEYMOD_LEN, DMA_TO_DEVICE);
 	kfree(encapdesc);
+	kfree(lkeymod);
 
 	return retval;
 }
@@ -971,6 +972,7 @@ int sm_keystore_slot_import(struct device *dev, u32 unit, u32 slot, u8 keycolor,
 			 DMA_TO_DEVICE);
 	dma_unmap_single(dev, keymod_dma, SECMEM_KEYMOD_LEN, DMA_TO_DEVICE);
 	kfree(decapdesc);
+	kfree(lkeymod);
 
 	return retval;
 }
-- 
1.7.9.5

